libethcore/SealEngine.cpp

```diff
@@ -43,9 +43,11 @@ void SealEngineFace::populateFromParent(BlockHeader& _bi, BlockHeader const& _pa
 	_bi.populateFromParent(_parent);
 }
 
-void SealEngineFace::verifyTransaction(ImportRequirements::value _ir, TransactionBase const& _t, BlockHeader const&) const
+void SealEngineFace::verifyTransaction(ImportRequirements::value _ir, TransactionBase const& _t, BlockHeader const& _bi) const
 {
-	if (_ir & ImportRequirements::TransactionSignatures)
+	if ((_ir & ImportRequirements::TransactionSignatures) && _bi.number() < chainParams().u256Param("metropolisForkBlock") && _t.hasZeroSignature())
```

Comment:
Previously zero signature before metropolis was rejected in TransactionBase constructor during `m_vrs.isValid()` check.

This new check here is needed because now zero signature is always accepted in TransactionBase constructor without getting to `m_vrs.isValid()` check.
So without this new check it would be consensus change before metropolis (consensus about zero signature transactions)